toucan_connectors/toucan_connector.py

```diff
@@ -468,3 +469,37 @@ def get_identifier(self):
 
     def describe(self, data_source: ToucanDataSource):
         """ """
+
+
+TableInfo = Dict[str, Union[str, List[Dict[str, str]]]]
+
+
+class DiscoverableConnector(ABC):
+    @abstractmethod
+    def get_model(self) -> List[TableInfo]:
+        """Tries to get all tables that are available for the connector"""
+        pass
+
+    @abstractmethod
+    def get_model_with_info(self) -> Tuple[List[TableInfo], Dict]:
+        """Tries to get all tables that are available for the connector and gives back some infos about what happenned"""
+        return (self.get_model(), dict())
+
+    @staticmethod
+    def format_db_model(
```

Comment:
I don't really see the point of switching from a plain method to an abstract class method. Not that important though

Reply:
This method has always been used to format only `get_model()`. 

So, in creating a discoverable connector, I thought it would make sense to bring it all together.